sdl2-compat: add setup hook and minor fixes

marcin-serwin opened the pull request:
This adds a setup hook analogous to the one used by SDL2 package that
picks up satelitte libraries during build time. This makes it possible
to replace SDL2 with sdl2-compat in packages using satelitte libraries
without further changes to build flags.
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

marcin-serwin committed:
sdl2-compat: add setup hook

This adds a setup hook analogous to the one used by SDL2 package that
picks up satelitte libraries during build time. This makes it possible
to replace SDL2 with sdl2-compat in packages using satelitte libraries
without further changes to build flags.

marcin-serwin committed:
sdl2-compat: specify as compatible with all platforms

The SDL library has backends for many platforms, not just unix ones,
however, it's currently restricted by its dependencies.

nadiaholmquist commented:
Seems good to me. I wasn't exactly sure ehat the purpose of the setup hook for SDL2 was, and it seemed to work fine without, but we definitely need this if it helps with using sdl2-compat as a replacement for SDL2.

SuperSandro2000 commented on pkgs/by-name/sd/sdl2-compat/package.nix:
```diff
@@ -30,7 +29,6 @@ stdenv.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [
     cmake
-    ninja
```
cmake+ninja is usually way faster than just cmake+make.

SuperSandro2000 replied:
For some big applications it really makes a difference but regardless I would just leave it here as we plan to use ninja for everything cmake related anyway.

LordGrimmauld commented on pkgs/by-name/sd/sdl2-compat/package.nix:
```diff
@@ -92,7 +93,7 @@ stdenv.mkDerivation (finalAttrs: {
     license = lib.licenses.zlib;
     maintainers = with lib.maintainers; [ nadiaholmquist ];
-    platforms = lib.platforms.unix;
+    platforms = lib.platforms.all;
```
did you test this, especially on darwin?

LordGrimmauld replied:
ah, good point. If it was previously unix it should be fine.

marcin-serwin committed:
sdl2-compat: skip slow tests

The slow tests rely on timers executing within a specified time which is not necessarily true if system is under a heavy load [1]. These tests are also disabled in the official CI pipeline [2].

[1]: https://github.com/NixOS/nixpkgs/pull/386759#issuecomment-2707779101
[2]: https://github.com/libsdl-org/sdl2-compat/blob/9544651fb39deefb2dd1aed9113af3b60898a1bb/.github/workflows/main.yml#L116
